mesonbuild/dependencies.py

```diff
@@ -1599,6 +1600,117 @@ def __init__(self, environment, kwargs):
     def get_link_args(self):
         return []
 
+class LLVMDependency(Dependency):
+    """LLVM dependency.
+
+    LLVM uses a special tool, llvm-config, which has arguments for getting
+    c args, cxx args, and ldargs as well as version.
+    """
+
+    # Ordered list of llvm-config binaries to try. Start with base, then try
+    # newest back to oldest (3.5 is abitrary), and finally the devel version.
+    llvm_config_bins = [
+        'llvm-config', 'llvm-config-4.0', 'llvm-config-3.9', 'llvm-config39',
+        'llvm-config-3.8', 'llvm-config38', 'llvm-config-3.7', 'llvm-config37',
+        'llvm-config-3.6', 'llvm-config36', 'llvm-config-3.5', 'llvm-config35',
+        'llvm-config-3.4', 'llvm-config34', 'llvm-config-3.3', 'llvm-config33',
+        'llvm-config-devel',
+    ]
+    llvmconfig = None
+    _llvmconfig_found = False
+    __best_found = None
+
+    def __init__(self, environment, kwargs):
+        super().__init__('llvm-config', kwargs)
+        self.cargs = []
+        self.libs = []
+        self.components = []
+
+        req_version = kwargs.get('version', None)
+        if self.llvmconfig is None:
+            self.check_llvmconfig(req_version)
+        if not self._llvmconfig_found:
+            if self.__best_found is not None:
+                mlog.log('found {!r} but need:'.format(self.version),
```

Comment:
This log message will end up on the next line after `mlog.log('Found llvm-config:', mlog.red('NO'))` from `self.check_llvmconfig()`, so it might not be clear that it refers to `llvm-config`. Maybe it's better to combine the two into one message—construct a list of arguments, and then pass that as `mlog.log(*msg)`.